### PR TITLE
Fix: Truncate long issue descriptions to prevent Salesforce overflow

### PR DESCRIPTION
**Pull request by jag-j**
## Summary
Fixes overflow issue where very long GitHub issue descriptions exceeded Salesforce's 32,768 character Long Text Area field limit, causing work item creation to fail with STRING_TOO_LONG errors.

## Changes
- Implemented truncation of GitHub issue bodies at 10,000 characters before HTML conversion
- Added MAX_MARKDOWN_LENGTH constant with safety margin for HTML expansion (~3x) and prepended text
- Truncated content includes helpful message directing users to full issue on GitHub
- Added comprehensive test suite for formatToGus function covering:
  - Normal markdown formatting
  - Empty and null body handling
  - Truncation of content exceeding 10,000 chars
  - Preservation of content under the limit
  - Markdown formatting in truncated content

## Root Cause
The issue occurred when GitHub webhook events contained very long issue descriptions that were:
1. Converted from markdown to HTML (increasing size by 1.5x-3x)
2. Prepended with GitHub issue link and separators (~60 chars)
3. Inserted into Salesforce Long Text Area fields without validation
4. Rejected by Salesforce with STRING_TOO_LONG error when exceeding 32,768 character limit

## Test plan
- [x] Added unit tests covering truncation logic, edge cases, and markdown formatting
- [ ] Manual testing with very long issue descriptions (>10,000 chars)
- [ ] Verify Salesforce work items are created successfully with truncated content
- [ ] Confirm truncation message renders properly in GUS

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: api/actions/__test__/formatToGus.spec.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        // Result should be significantly shorter than the original after truncation
+        expect(result.length).toBeLessThan(longBody.length);
+    });
+
+    it('should keep content under 10,000 characters intact', () => {
+        const body = 'A'.repeat(9000);
+        const result = formatToGus(testUrl, body);
+
+        expect(result).not.toContain('Content truncated');
+        expect(result).toContain('A'.repeat(100)); // Sample check
```
Comment (jfeingold35) [inline api/actions/__test__/formatToGus.spec.js:53]: Is there a reason that you're not doing `expect(result).toContain(body)`? If it's checking that the result wasn't truncated, then it should check for the untruncated body, right? At the very least, should it check that there are 9000 `A`'s in the output?
